infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -225,40 +228,47 @@ private void waitMachines(
    * start of servers probes.
    */
   private Function<Void, CompletionStage<Void>> checkServers(
-      List<CompletableFuture<?>> toCancelFutures, KubernetesMachine machine) {
+      List<CompletableFuture<?>> toCancelFutures,
+      KubernetesMachine machine,
+      EnvironmentContext context) {
     return ignored -> {
-      // This completable future is used to unity the servers checks and start of probes
-      final CompletableFuture<Void> serversAndProbesFuture = new CompletableFuture<>();
-      final String machineName = machine.getName();
-      final RuntimeIdentity runtimeId = getContext().getIdentity();
-      final ServersChecker serverCheck =
-          serverCheckerFactory.create(runtimeId, machineName, machine.getServers());
-      final CompletableFuture<?> serversReadyFuture;
       try {
-        serversReadyFuture = serverCheck.startAsync(new ServerReadinessHandler(machineName));
-        toCancelFutures.add(serversReadyFuture);
-        serversAndProbesFuture.whenComplete((ok, ex) -> serversReadyFuture.cancel(true));
-      } catch (InfrastructureException ex) {
-        serversAndProbesFuture.completeExceptionally(ex);
+        EnvironmentContext.setCurrent(context);
```

Comment:
What if there would be a method that wraps anything with context setting. Would the code be simpler with less indentation (which leads to complexity of code reading)?
Example:
```java
.thenCompose(setContext(context, bootstrap(toCancelFutures, machine)))
```

Reply:
sure, that makes sense, I'll rework that